text/0000-sigstore-integration.md

```diff
@@ -0,0 +1,387 @@
+
+However, in practice most OSS projects where there are more than a handful of contributors will very likely use CI/CD (Build) capabilities provided through an aaS model. In this scenario, identity in the context of the build activity now shifts to the NPE model. Here, providers capable of authenticating to the public Sigstore service via organic OIDC capabilities. Mainstream commercial build plants running aaS are de-facto subject to various standards and certifications (both operational and security), and so can provide certain levels of attestation metadata related to the production of the artifact. Further, they of course have no privacy concerns. It is for these reasons, and others that the team at GH implementing Sigstore tooling for the NPM ecosystem have chosen to focus on the aaS build scenarios as they talk with and evangelize that focus as a broader pattern. This is a pattern that this RFC agrees with and adopts.
+
+## Authentication
+
+Sigstore uses the OIDC pattern and ecosystem as the basis for authentication against the signing issuer (Fulcio). We assume that most are familiar with the OIDC pattern and tooling and so will not address those topics here, with the exception of the OIDC identity payload attributes. The certificates used for signing of course need to have an associated identity:
+
+* In the case of an individual developer signing from his desktop, Sigstore will embed the email address of the signing individual in the metadata delivered to the Rekor service. 
+* In the case of an NPE signer the metadata delivered to the Rekor service will have an Issuer attribute that looks something like this: "Issuer": "https://token.actions.githubusercontent.com".
+
+## Authorization
+
+With respect to AuthZ, while there has been discussion in the pre-rfc thread regarding TUF, this proposal asserts that two phases will need to be undertaken. Phase one will rely on the existing mechanisms implemented on top of git and provided by the most commonly used platforms (GitHub, GitLab, etc, etc) that assign OSS project participants to a group, i.e. “Maintainers” or “Members” in the case of GH. Only project maintainers are typically allowed to either trigger a build event manually or implement automation to do so based on a set of project specific evaluation criteria. We choose this approach to AuthZ as it will allow focus on the core Sigstore implementation for cargo and crates.io in the near term, and a deliberate focus on TUF in the next phase. This is important as TUF is tricky to implement well and represents a significant shift from the existing AuthZ approach that is heavily socialized/embedded broadly in the OSS ecosystem currently.
```

Comment:
This is confusing to me since only people with direct access to publish a crate have access to do so on crates.io. The authors field and the actual people with access to publish it are disjoint. I'm unsure why this distinction is important; members just don't have to be part of the protocol.

Additionally, this mentions AuthZ and TUF without explaining them at all, and should.